src/mjolnir/elevationbuilder.cc

```diff
@@ -167,69 +206,46 @@ void add_elevation(const boost::property_tree::ptree& pt,
 namespace valhalla {
 namespace mjolnir {
 
-void ElevationBuilder::Build(const boost::property_tree::ptree& pt) {
-
-  auto hierarchy_properties = pt.get_child("mjolnir");
-  std::string tile_dir = hierarchy_properties.get<std::string>("tile_dir");
-
-  // Crack open some elevation data if its there. Return if it is not.
+void ElevationBuilder::Build(const boost::property_tree::ptree& pt, const std::string& tile) {
```

Comment:
In my view it's the binary's responsibility to parse tileId string to tileId GraphId. I think it's better to have collection of graphIds for example
```suggestion
void ElevationBuilder::Build(const boost::property_tree::ptree& pt, std::deque<GraphId> tiles) {
```

Reply:
for now ElevationBuilder has everything which is needed for this job more over it does this already

cons of this proposal:
1. unnecessary complexity in stand alone binary (additional work)
2. if the logic of worknig with tilesets would change we would need to change in at least 2 places, in binary and in the current Build implementation
3. additional public method(not so negative effect but still additional code)

pros
1. can pass all tiles in one call(instead of calling Build in a loop)

Reply:
i was thinking it would be all config driven. if nothing is provided in the config it uses the `tile_dir` to find out what graph tiles are in the tileset and gets elevation data for them. otherwise we can add a key to the `additional_data` part of the config and call it `elevation_shape`. its value can be the path to a file on the filesystem, it could point to a geojson file, it could point to a file with a list of tiles in it, it could point to whatever, but then its all configuration based and again then the `valhalla_add_elevation` can be made very stupid and all the logic can live inside of `Build`. thats just my opinion though.

Reply:
it seems to me you guys want to have to tell the Build which tiles to get elevation for rather than build figure that out on its own. why is that? i cant understand the use case. the only thing that that allows you to add is the ability to only add elevation to a subset of tiles in your tileset. i suppose this could be useful if you had more than one elevation data source OR! if you were trying to distribute the processing so that you could do parts of the tileset in parallel on multiple machines.... i guess its the latter one then :smile:  ok cool, you guys should have just mentioned that up front, it makes it much easier to understand the motivations haha apologies if i missed that earlier. so yeah it makes sense here. i think passing a file with the list of tiles in it too the valhalla_add_elevation binary and passing a `deque` to Build makes sense. we just need to update the other calls to Build to compute the tileset first before calling, would it be ok to change it to an `unordered_set` since graphreader.gettileset already returns that type?

Reply:
you're kinda right @kevinkreiser . My idea was to use it in the distributed or/and "cache-able" pipeline when you split planet tileset into the batches(e.g 1x1 degree squres, or 2x2 degree, or just single tile in a batch) (see pseudo code in the [comment](https://github.com/valhalla/valhalla/pull/3391#issuecomment-965057330))
- process batches sequentially and cache result of the batch not to lose progress
- process batches in parallel so caching might not be even needed
Yeah, this is not a secret at all :) we should've mentioned it somewhere in the PR description